openghg/store/_flux_timeseries.py

```diff
@@ -0,0 +1,273 @@
+            check_if_need_new_version,
+        )
+
+        species = clean_string(species)
+        source = clean_string(source)
+        if domain:
+            domain = clean_string(domain)
+
+        if overwrite and if_exists == "auto":
+            logger.warning(
+                "Overwrite flag is deprecated in preference to `if_exists` (and `save_current`) inputs."
+                "See documentation for details of these inputs and options."
+            )
+            if_exists = "new"
+
+        # Making sure new version will be created by default if force keyword is included.
+        if force and if_exists == "auto":
+            if_exists = "new"
+
+        new_version = check_if_need_new_version(if_exists, save_current)
+
+        filepath = Path(filepath)
+
+        try:
+            source_format = FluxTimeseriesTypes[source_format.upper()].value
+        except KeyError:
+            raise ValueError(f"Unknown data type {source_format} selected.")
+
+        # Load the data retrieve object
+        parser_fn = load_flux_timeseries_parser(source_format=source_format)
+
+        _, unseen_hashes = self.check_hashes(filepaths=filepath, force=force)
+
+        if not unseen_hashes:
+            return {}
+
+        filepath = next(iter(unseen_hashes.values()))
+
+        # Define parameters to pass to the parser function
+        # TODO: Update this to match against inputs for parser function.
+        param = {
+            "filepath": filepath,
+            "species": species,
+            "region": region,
+            "source": source,
+            "data_type": "flux_timeseries",
+            "period": period,
+            "continuous": continuous,
+        }
+
+        optional_keywords: dict[Any, Any] = {
+            "database": database,
+            "database_version": database_version,
+            "model": model,
+        }
+
+        signature = inspect.signature(parser_fn)
+        fn_accepted_parameters = [param.name for param in signature.parameters.values()]
+
+        input_parameters: dict[Any, Any] = param.copy()
+
+        # Checks if optional parameters are present in function call and includes them else ignores its inclusion in input_parameters.
+        for param, param_value in optional_keywords.items():
+            if param in fn_accepted_parameters:
+                input_parameters[param] = param_value
+            else:
+                logger.warning(
+                    f"Input: '{param}' (value: {param_value}) is not being used as part of the standardisation process."
+                    f"This is not accepted by the current standardisation function: {parser_fn}"
+                )
+
+        flux_timeseries_data = parser_fn(**input_parameters)
+
+        # Checking against expected format for Flux
+        for split_data in flux_timeseries_data.values():
+            em_data = split_data["data"]
+            FluxTimeseries.validate_data(em_data)
+
+        min_required = ["species", "source"]
```

Comment:
Also just noticed, we'd likely need the required keys to also include `region` as we will always need to define which area the flux is related to when storing this (c.f. `domain` for `flux` inputs).